### ISSUE TITLE
Server-side render

### ISSUE THREAD

**Issue by NotIntMan** [creationix/domchanger#11]
Hello. Your library looks stunning minimalist. I would like to try to use it in one of my works, but I need to be able to get a static html-code as a string on the server. Is it possible?

**Comment by creationix** [comment]
It would be pretty easy to reuse my JSON-ML format and generate an html string directly from it.

I don't know how much of the actual code here would be useful for that task though.

If you could write up an example of how you would like it to work on the server, I could whip up an implementation for you.

**Comment by NotIntMan** [comment]
Oh, interface for that task is very simple. Just a function that takes a component and its state to produce the code. Something like this:
```js
function HelloMessage() {
  return { render: render };
  function render(name) {
    return ["div", "Hello " + name];
  }
}

// function toHTML(c: Component, state: Object): string;
// or, for generic-like style
// function toHTML<T>(c: Component<T>, state: T): string;
var html = domChanger.toHTML(HelloMessage, "Tim");
```

**Comment by creationix** [comment]
Are you sharing components between the server and client or will this be server only?

**Comment by creationix** [comment]
For pure server-side, I like to use just dombuilder and plain functions for composition.

https://github.com/creationix/dombuilder

Go ahead and read the tiny source for that.  It would be pretty trivial to fork it to output an HTML string instead of creating a dom tree.

**Comment by NotIntMan** [comment]
No, it's for sharing components between client and server. I think i'm already found the solution, but, please, implement this feature. This is *must have* feature for library like this.

**Comment by creationix** [comment]
Ok, sounds good.  Ping me in a day or so if I haven't gotten to it yet, my backlog is pretty full these days.

**Comment by NotIntMan** [comment]
Hello again, @creationix! Do you still want to do it?:)

**Comment by creationix** [comment]
![image](https://user-images.githubusercontent.com/89353/43339505-1d986a60-919f-11e8-86b4-52f5451876b7.png)

Yes, but I'm drowning in Github notifications.  I'll see if I can work on it right now since it might be related to my work stuff.  If I postpone it I'll forget.

**Comment by creationix** [comment]
Hmm, actually I see this is for domchanger.  I don't think I'll have time to make it work server-side like the other frameworks do.  I was thinking this was adding HTML output to dombuilder which is a much easier task.
